.github/workflows/ci.yml

```diff
@@ -332,15 +332,11 @@ jobs:
       - set-condition
       - sequentialise
     steps:
+      - uses: actions/checkout@v4
```

Comment:
should we always add `fetch-depth: 0` here?

Reply:
I don't think this is as important

Reply:
It can save some disk space.

Reply:
That's true.

Having `fetch-depth` will require more space - it explains why we don't need to add it if not required